### DISCUSSION TITLE
Select - Fire event when popover is closed

### DISCUSSION THREAD

**Discussion by binhxn** [ariakit/ariakit#1780] [Questions] [answered]
Hello! 👋 

I was wondering if it's possible to fire some event when SelectPopover is closed when clicking outside of it? Kind of like a `onPopoverClosed` callback. Is it possible to do something like that? For example:

```jsx
<SelectPopover onPopoverClose={() => console.log('event fired!')}
```

I tried creating a condition using the `mounted` and `open` property from `useSelectState` hook, but that seems to trigger the event when the page loads, rather than when closing the popover. Thoughts on how to go about this?

**Comment by diegohaz** [comment, marked as answer]
If you want an event that fires only when clicking outside, you can use the `hideOnInteractOutside` prop:

```jsx
<SelectPopover
  hideOnInteractOutside={() => {
    onClickOutside();
    return true;
  }}
/>
```

If you want to do something whenever the popover closes, no matter the reason:

```jsx
const select = useSelectState({
  setOpen: (open) => {
    if (select.open && !open) {
      onClose();
    }
  }
});
```

**Reply by binhxn** [reply]
Awesome, these handle my use cases. Thank you for the response !
